include/pybind11/pytypes.h

```diff
@@ -183,7 +183,7 @@ class object_api : public pyobject_tag {
     str_attr_accessor doc() const;
 
     /// Return the object's current reference count
-    int ref_count() const { return static_cast<int>(Py_REFCNT(derived().ptr())); }
+    ssize_t ref_count() const { return Py_REFCNT(derived().ptr()); }
```

Comment:
This was a bug - Free-threaded Python sets this to `UINT32_MAX` for immortal objects, which is larger than an `int` can hold. Changed the return type to match CPython's `Py_ssize_t`.

Reply:
I left the old behavior there for PyPy - stripping the top bits was needed because PyPy uses a couple of them for it's own things.